mantidproject/mantid
# Allow input workspaces to have bank ID 0 in fitting tabs of the Engineering GUI

A bank ID of 0 is now assigned to the output workspace of EnggFocus if no bank is specified (ie the focus is cropped). The size of RunMap was increased by 1 to allow for this. Now focused runs with a bank ID of 0 can be loaded into the Fitting and GSAS tabs of the GUI and processed as normal.

**Description of work**

**To test:**

Run a focus. To do this, go to the GUI (Interfaces -> Diffraction -> Engineering Diffraction).

You'll need to run a calibration first - go to the Calibration tab, and under **Make new calibration**, enter 236516 in **Vanadium #** and 241391 in **Calibration Sample #**. Click **Calibrate** (this might take a while!).

Now go to the Focus tab, and select [this file](https://github.com/mantidproject/mantid/files/1964232/ENGINX00281188.zip). Tick the **Focus Cropped** box, and choose some spectrum numbers (something like 100-200, but not that since I already tried it with that). Go and find the saved Nexus file (it'll have been saved somewhere annoying - this is being fixed in #22361). It's probably easiest to do `find -name "ENGINX_281188_focused_cropped.nxs" -print` from your build directory.

Load this into the Fitting tab. Click on its label in the list next to the plot to plot it - the plot should appear with no problems. Do the same in the GSAS tab

Fixes #22376 

**Release Notes** 

Scientists aren't using a version of Mantid recent enough to have this bug in, so it doesn't need to go in the release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
